README.md

```diff
@@ -1,2 +1,88 @@
 # app.js
-GitHub Apps toolset for Node.js
+
+> Official GitHub Platform for GitHub Apps in Node.js
+
+## GitHub Apps
+
+`@octokit/app.js` allows to authenticate as GitHub Apps, installations, listen
+to webhooks, and respond to them using the GitHub API.
+
+### Authenticating as an App
+
+In order to authenticate as a GitHub App, you need to generate a Private Key
+and use it to sign a JSON Web Token (jwt) and encode it. 
+
+```js
+const App = require('@octokit/app.js')
+
+const app = new App({id, privateKey})
+const appAuth = await app.appAuth()
+
+// Example of using authetincated app to GET an individual installation
+// https://developer.github.com/v3/apps/#find-installations
+const installation = await appAuth.request({
+  method: 'GET',
+  url: '/repos/:owner/:repo/installation',
+  headers: { accept: 'application/vnd.github.machine-man-preview+json' },
+  owner: 'hiimbex',
+  repo: 'testing-things'
+})
+
+// contains the installation id necessary to authenticate as an installation
+const installationId = installation.data.id
+```
+
+### Authenticating as an Installation
+
+Once you have authenticated as a GitHub App, `auth` here, you can use that
+in order to request an installation access token. This token is scoped for
+your specific app and expires after an hour.
+
+```js
+const app = require('@octokit/app.js')
+
+const app = new App({id, privateKey})
+const installation = await app.requestToken(app.appAuth(), {installationId: 123})
+
+// Example of opening an issue as an installation
+// https://developer.github.com/v3/issues/#create-an-issue
+const result = await installation.issues.create({
+  owner: 'hiimbex',
+  repo: 'tetsing-things',
+  title: 'My installation's first issue!'
+})
```

Comment:
This is really interesting! I didn’t think of getting a client API from `app.appAuth()` or `app.requestToken()`, I thought they would only return the tokens. 

I’m not sure what the better approach is here. 

The advantage of only returning the token is that the `@octokit/app` package will be much simpler and less opinionated. The user can use their own request library. I was thinking of something like this instead:

```js
const got = require('got') // could be any request library really
const app = require('@octokit/app.js')
const installationAccessToken = await app.requestToken({installationId: 123})

// https://developer.github.com/v3/issues/#create-an-issue
const {data: {total_count}} = await got.post('https://api.github.com/repos/hiimbex/tetsing-things/issues', {
  body: {title: 'My installation's first issue'},
  headers: {
    authorization: `token ${installationAccessToken}`,
    accept: 'application/vnd.github.machine-man-preview+json'
  }
})
```

I’m curious to hear what you think! 

Maybe as contrast, I really like your approach for the  "all-batteries-included" `octokit` npm package. The `octokit` package would have an `.app()` method which would use `@octokit/app` internally, but it would also have all the REST and GraphQL APIs etc. I could imagine an API like this

```js
const Octokit = require('octokit')
const client = new Octokit()

const app = client.app({id, privateKey})
const installation = app.findRepoInstallation({owner, repo})
```

We wouldn’t even need to call a method like `app.appAuth()` to get the JWT token, it could all happen internally when needed, because we have the app id and privateKey as internal state. When the user calls `.findRepoInstallation()` than the `app` instance could check if it has a cached JWT token that didn’t time out yet. If it does, it uses that token to send the actual request. Otherwise it will send the request to receive the JWT token first, cache it for future requests, and only then send the actual request to create the issue.